### PR TITLE
Use signalling in datepicker banner service to improve load

### PR DESCRIPTION
**Pull request by oliverguenther**
- [x] Introduce a helper to load a signalled endpoint and get a custom, plain JSON object back
- [x] Use that signalled endpoint for loading only the IDs of related work packages

https://community.openproject.org/wp/42231

### CODE REVIEW

File: frontend/src/app/shared/components/datepicker/datepicker.modal.service.ts [hunk 1]
```diff
@@ -46,32 +46,36 @@ import {
...
     .work_packages
-    .filtered(ApiV3Filter('follows', '=', [this.changeset.id]))
-    .get()
+    .signalled(
+      ApiV3Filter('follows', '=', [this.changeset.id]),
+      ['elements/id'],
+    )
     .pipe(
-      map((collection) => collection.elements),
+      map((collection:IHALCollection<{ id:string }>) => collection._embedded.elements || []),
```
Comment (b12f) [inline frontend/src/app/shared/components/datepicker/datepicker.modal.service.ts:78]: Am I right in thinking that this returns a signalled `HalResource`? e.g. `{ id:string }` is a WorkPackageResource with half the stuff set? Or is this a plain JS object?

Comment (oliverguenther) [reply frontend/src/app/shared/components/datepicker/datepicker.modal.service.ts:78]: Just a plain JS object, as we're not going through the HalResourceService. I added this to hopefully make it a bit easier to use signalled/plain objects in the APIv3 service without going through the HalResource classes.

Comment (b12f) [reply frontend/src/app/shared/components/datepicker/datepicker.modal.service.ts:78]: Perhaps the next step would be to use these same APIs everywhere we use signalling

**Review by b12f — APPROVED** [review]
LGTM

File: frontend/src/app/shared/components/datepicker/datepicker.modal.service.ts (hunk 1 above)
Comment (oliverguenther) [reply frontend/src/app/shared/components/datepicker/datepicker.modal.service.ts:78]: Yes, that's my plan and hope as well! :D

**Pull request merged**